test/unit-luatest/config_test.lua

```diff
@@ -0,0 +1,44 @@
+local t = require('luatest')
+local lcfg = require('vshard.cfg')
+local uuid = require('uuid')
+
+local g = t.group('config')
+
+g.test_basic_uri = function()
+    local url = 'storage:storage@127.0.0.1:3301'
+    local storage_1_a = {
+        uri = url,
+        name = 'storage_1_a',
+    }
+    local config = {
+        sharding = {
+            storage_1_uuid = {
+                replicas = {
+                    storage_1_a_uuid = storage_1_a,
+                }
+            },
+        },
+    }
+    t.assert(lcfg.check(config), 'normal uri')
+
+    storage_1_a.uri = {url}
+    t.assert(lcfg.check(config), 'table uri')
+
+    storage_1_a.uri = {url, params = {transport = 'plain'}}
+    t.assert(lcfg.check(config), 'uri with options')
+
+    storage_1_a.uri = 3301
+    t.assert(lcfg.check(config), 'number uri')
+
+    storage_1_a.uri = 'bad uri ###'
+    t.assert_error(lcfg.check, config)
+
+    storage_1_a.uri = {}
+    t.assert_error(lcfg.check, config)
+
+    storage_1_a.uri = nil
+    t.assert_error(lcfg.check, config)
+
+    storage_1_a.uri = uuid.new()
+    t.assert_error(lcfg.check, config)
```

Comment:
I suggest to add "empty string" case. Since error message explicitly says about it.

Reply:
Done.